src/main/java/io/dashboardhub/pipelinedashboard/service/ProjectService.java

```diff
@@ -1,12 +1,71 @@
 package io.dashboardhub.pipelinedashboard.service;
 
+import io.dashboardhub.pipelinedashboard.domain.Project;
+import io.dashboardhub.pipelinedashboard.domain.User;
 import io.dashboardhub.pipelinedashboard.repository.ProjectRepository;
+import io.dashboardhub.pipelinedashboard.service.exception.PermissionDeniedException;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.context.SecurityContextHolder;
 import org.springframework.stereotype.Service;
 
 @Service
 public class ProjectService {
 
     @Autowired
-    ProjectRepository projectRepository;
+    private ProjectRepository projectRepository;
+
+    @Autowired
+    private UserService userService;
+
+    private String getCurrentUsername() {
```

Comment:
Actually an **Advice** already exists for the **CurrentUser**, it just needs updating with the full User object https://github.com/DashboardHub/PipelineDashboard/blob/prototype-v0.3/src/main/java/io/dashboardhub/pipelinedashboard/controller/advice/CurrentUserAdvice.java